pkg/bytecode/compiler.go

```diff
@@ -207,10 +219,10 @@ func (c *Compiler) compileDecl(decl *parser.Decl) error {
 }
 
 func (c *Compiler) compileAssignment(stmt *parser.AssignmentStmt) error {
-	if err := c.Compile(stmt.Target); err != nil {
```

Comment:
This is an unrelated bug fix is it? If so, please put it in a separate commit so it gets its own commit message rather than it being hidden and undescribed in an unrelated commit. This is useful when looking back on history to understand why a change was made, particularly bug fixes when you find only half a bug was fixed. You see this change in history and it has a completely unrelated description on it ("Implement arrays" ???).

This seems like a pretty fundamental bug. How come no test cases picked this up? Is there a test case added for this?

Or am I misunderstanding this because I'm not sure how assignment even worked if the target and value were backwards.

Reply:
Spoke offline, I have split the fix into a separate commit, and then a follow up commit to add the test to cover array concatenation not manipulating the original arrays, which doesn't pass without the fix.